chia/rpc/wallet_rpc_api.py

```diff
@@ -1706,17 +1706,37 @@ async def nft_get_info(self, request: Dict) -> EndpointResult:
                     "success": False,
                     "error": f"Launcher coin record 0x{uncurried_nft.singleton_launcher_id.hex()} not found",
                 }
+            # Get minter DID
+            eve_coin = (
+                await self.service.wallet_state_manager.wallet_node.fetch_children(
+                    launcher_coin[0].coin.name(), peer=peer
+                )
+            )[0]
+            eve_coin_spend: CoinSpend = await self.service.wallet_state_manager.wallet_node.fetch_puzzle_solution(
+                eve_coin.spent_height, eve_coin.coin, peer
+            )
+            eve_full_puzzle: Program = Program.from_bytes(bytes(eve_coin_spend.puzzle_reveal))
+            eve_uncurried_nft: Optional[UncurriedNFT] = UncurriedNFT.uncurry(*eve_full_puzzle.uncurry())
+            if eve_uncurried_nft is None:
+                return {"success": False, "error": "The coin is not a NFT."}
+            minter_did = None
+            if eve_uncurried_nft.supports_did:
```

Comment:
`supports_did` for eve and current coin should be the same, so you can move the fetching bit above under this `if` too